internal/versions/versions.go

```diff
@@ -192,13 +192,21 @@ func (h *handler) GetOCPSemVer(openshiftVersion string) (string, error) {
 }
 
 func (h *handler) AddOpenshiftVersion(ocpReleaseImage, pullSecret string) (*models.OpenshiftVersion, error) {
+	// Check whether ocpReleaseImage already exists in cache
+	for _, version := range h.openshiftVersions {
+		if version.ReleaseImage != nil && *version.ReleaseImage == ocpReleaseImage {
+			// Return existing version
+			return &version, nil
```

Comment:
I think this is @mhrivnak's favorite bug.

https://www.evanjones.ca/go-gotcha-loop-variables.html

Reply:
Although it might work because it's a return in this case.
Still makes me uneasy.

Reply:
Yeah, I don't think it's a problem here. You could argue that returning `&h.openshiftVersions[i]` would let `version` get garbage-collected slightly sooner, but that level of optimization is unlikely to have even a measurable effect.

Reply:
Right, shouldn't be an issue here, but I'll just return a copy of the value to make it cleaner :)

Reply:
> Still makes me uneasy.

That was more of an academic concern around how easy it would be to introduce the bug with a refactoring, but yeah, this is fine however. Sorry for the noise :sweat_smile: